04_numeronym/n0npax/main_test.go

```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+	"bytes"
+	"strconv"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMainOutput(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+
+	main()
+
+	expected := strconv.Quote("[K8s anz i18n]\n")
+	actual := strconv.Quote(buf.String())
+
+	if expected != actual {
+		t.Errorf("Unexpected output. Expected: %q - Actual: %q", expected, actual)
+	}
+}
+
+func TestNumeronyms(t *testing.T) {
+	actual := numeronyms("K8bernetes", "anz", "internationalization")
+	expected := []string{"K8s", "anz", "i18n"}
+	for i, v := range actual {
+		if expected[i] != v {
+			t.Errorf("Unexpected output. Expected: %q - Actual: %q", expected[i], v)
+		}
+	}
+}
+
+func TestShorten(t *testing.T) {
+	testCases := map[string]struct {
+		input    string
+		expected string
+	}{
+		"anz":                  {input: "anz", expected: "anz"},
+		"Kubernetes":           {input: "Kubernetes", expected: "K8s"},
+		"internationalization": {input: "internationalization", expected: "i18n"},
+	}
+	for name, test := range testCases {
+		input, want := test.input, test.expected
+		t.Run(name, func(t *testing.T) {
+			actual := shorten(input)
+			assert.Equalf(t, want, actual, "For input: %v expected: %v got %v", input, want, actual)
```

Comment:
Again, don't feel the need to change this.
You can avoid the need to pass `t` into these `testify` functions by creating an assertions object at the start of the test
```
func TestName(t *testing.T) {
  a := assert.New(t)
  // Test Code
  a.Equalf(want, actual, "message") // Don't need to pass t
}
```

This is really a matter of preference, you trade having to pass `t` for having to create `a` at the start.

Reply:
Thanks for pointing this out 👍